src/vue-floating-ui.ts

```diff
@@ -60,36 +60,34 @@ export function useFloating({
 }: UseFloatingProps = {}): UseFloatingReturn {
 	const reference = ref<ReferenceElement | null>(null);
 	const floating = ref<FloatingElement | null>(null);
+	const middlewareRef = ref(middleware);
+	const cleanupRef = ref<Function | void | null>(null);
 	// Setting these to `null` will allow the consumer to determine if
 	// `computePosition()` has run yet
-	const x = ref<number | null>(null);
-	const y = ref<number | null>(null);
-	const _placement = ref(placement);
-	const _strategy = ref(strategy);
-	const _middleware = ref(middleware);
-	const middlewareData = ref<MiddlewareData>({});
-	const cleanupRef = ref<Function | void | null>(null);
+	const data = reactive<Data>({
+		x: null,
+		y: null,
+		placement: unref(placement),
+		strategy: unref(strategy),
```

Comment:
> The reactive conversion is "deep": it affects all nested properties. A reactive object also deeply unwraps any properties that are [refs](https://vuejs.org/api/reactivity-core.html#ref) while maintaining reactivity.
> – https://vuejs.org/api/reactivity-core.html#reactive

`unref` can be omitted

Reply:
I think the key part is _"`unwraps` while maintaining reactivity_", we don't want to keep reactivity or it would mutate original props refs as before. We just want o use its value like an initial value. 

[Codesandbox](https://codesandbox.io/s/billowing-dew-tpp97f?file=/src/useSomething.js)